builder/docker/config.go

```diff
@@ -23,21 +23,22 @@ type Config struct {
 	common.PackerConfig `mapstructure:",squash"`
 	Comm                communicator.Config `mapstructure:",squash"`
 
-	Author         string
-	Changes        []string
-	Commit         bool
-	ContainerDir   string `mapstructure:"container_dir"`
-	Discard        bool
-	ExecUser       string `mapstructure:"exec_user"`
-	ExportPath     string `mapstructure:"export_path"`
-	Image          string
-	Message        string
-	Privileged     bool `mapstructure:"privileged"`
-	Pty            bool
-	Pull           bool
-	RunCommand     []string `mapstructure:"run_command"`
-	Volumes        map[string]string
-	FixUploadOwner bool `mapstructure:"fix_upload_owner"`
+	Author           string
+	Changes          []string
+	Commit           bool
+	ContainerDir     string `mapstructure:"container_dir"`
+	Discard          bool
+	ExecUser         string `mapstructure:"exec_user"`
+	ExportPath       string `mapstructure:"export_path"`
+	Image            string
+	Message          string
+	Privileged       bool `mapstructure:"privileged"`
+	Pty              bool
+	Pull             bool
+	RunCommand       []string `mapstructure:"run_command"`
+	Volumes          map[string]string
+	FixUploadOwner   bool `mapstructure:"fix_upload_owner"`
+	WindowsContainer bool `mapstructure:"windows_container"`
```

Comment:
Here, may be this could be a pointer var ?
```suggestion
	WindowsContainer *bool `mapstructure:"windows_container"`
```
This way when WindowsContainer is not set we guess it based on the current os ? using `if runtime.GOOS == "windows"`